File: pset6/readability/readability.py
# import get string function from CS50 library
from cs50 import get_string
import string

# ask user for a text to be analyzed
text = get_string("Text: ")

# create list of punctuation to check for below
punctuation = [".", "!", "?"]

# initialize letter and sentence count
letters = 0
sentences = 0

# split text by white space then take the lenght of the resulting list...this seems like an easier way to get the word count than counting spaces, with counting spaces you have to initialize at 1
words = len(text.split(" "))

# iterate over chars in string, incrementing letter if they are a letter, and sentences if they are a puncuation
for char in text:
    if char.isalpha():
        letters += 1
    if char in punctuation:
        sentences += 1

# calculate L and S...using floats for precision
L = float(letters) / float(words) * 100
S = float(sentences) / float(words) * 100

# forumla for Coleman-Liau index
Grade = round(0.0588 * L - 0.296 * S - 15.8)

# if below 1, print before grade 1, if above print 16+, elsewise print grade level
if Grade < 1:
    print("Before Grade 1")

elif Grade > 16:
    print("Grade 16+")

else:
    print(f"Grade {Grade}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //get input from user using CS50 provided function
    string text = get_string("Text: ");
    //create variables to keep track of the number of letters, words, and sentences
    int letters = 0;
    //initiating at 1, because it won't count the first word with this algorithm
    int words = 1;
    int sentences = 0;
    //for loop to iterate through text and increment the counts depending on the character
    for (int i = 0; i < strlen(text); i++)
    {
        //check for letters
        if (isalpha(text[i]) > 0)
        {
            letters++;
        }
        //check for spaces i.e. words
        if (isspace(text[ i + 1]) > 0)
        {
            words++;
        }
        //check for new sentences by looking for punction...each number is a ?, !, or .
        if (text[i] == 33 || text[i] == 63 || text[i] == 46)
        {
            sentences++;
        }
    }

    float L = ((float) letters) / ((float) words) * 100;
    float S = ((float) sentences) / ((float) words) * 100;;

    float index =  0.0588 * L - 0.296 * S - 15.8;

//prints out the grade level, if it is below first grade or above sixteenth, it prints that out instead (as per CS50 specifications)
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}